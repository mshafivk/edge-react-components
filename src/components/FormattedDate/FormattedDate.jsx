import React from 'react';
import PropTypes from 'prop-types';
import Moment from 'react-moment';

const formats = {
  fullDateTime: 'MMMM Do YYYY, h:mm:ss a',
  dateOnly: 'DD/MM/YYYY',
  fromNow: 'fromNow',
};

function FormattedDate({ format, children }) {
  const isFromNow = format === formats['fromNow'];
  return (
    <Moment format={!isFromNow && formats[format]} fromNow={isFromNow}>
      {children}
    </Moment>
  );
}

FormattedDate.propTypes = {
  format: PropTypes.string,
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.instanceOf(Date)])
    .isRequired,
};
FormattedDate.defaultProps = {
  format: 'fullDateTime',
};

export default FormattedDate;
