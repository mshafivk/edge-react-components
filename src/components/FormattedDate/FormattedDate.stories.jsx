import React from "react";
import FormattedDate from "./FormattedDate";

export default {
  title: "FormattedDate",
  component: FormattedDate,
};

const Tpl = () => <FormattedDate>2021-01-09T01:11:21Z</FormattedDate>;

const tableStyle = {
  borderCollapse: "collapse",
  border: "1px solid",
};
const rowStyle = {
  border: "1px solid",
};
const CustomTpl = () => (
  <table cellPadding="2" border="1" style={tableStyle}>
    <thead>
      <tr>
        <th>Format</th>
        <th>Example</th>
      </tr>
    </thead>
    <tbody>
      <tr style={rowStyle}>
        <td>fullDateTime</td>
        <td>
          <FormattedDate format={"fullDateTime"}>
            2021-01-09T01:11:21Z
          </FormattedDate>
        </td>
      </tr>
      <tr style={rowStyle}>
        <td>dateOnly</td>
        <td>
          <FormattedDate format={"dateOnly"}>
            2021-01-09T01:11:21Z
          </FormattedDate>
        </td>
      </tr>
    </tbody>
  </table>
);

export const CustomTemplate = CustomTpl.bind({});

export const FormattedDateDefault = Tpl.bind({});
FormattedDateDefault.args = {
  format: "MMM DD YYYY HH MM SS",
};

CustomTemplate.storyName = "Custom Formats";
FormattedDateDefault.storyName = "Default";
