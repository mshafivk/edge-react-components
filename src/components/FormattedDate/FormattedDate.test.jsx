import React from 'react';
import { mount } from 'enzyme';
import FormattedDate from './FormattedDate';

describe('Formatted Date Component', () => {
  test('should return Date only value if format is set as dateOnly', () => {
    const wrapper = mount(
      <FormattedDate format={'dateOnly'}>2021-01-09T01:11:21Z</FormattedDate>,
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('time').text()).toBe('09/01/2021');
  });

  test('should return Date and Time value if format is set as dateOnly', () => {
    const wrapper = mount(
      <FormattedDate format={'fullDateTime'}>
        2021-02-25T01:11:21Z
      </FormattedDate>,
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('time').text()).toBe('February 25th 2021, 6:41:21 am');
  });

  test('should return Date and Time value if format attribute is not passed', () => {
    const wrapper = mount(<FormattedDate>2021-02-25T01:11:21Z</FormattedDate>);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('time').text()).toBe('February 25th 2021, 6:41:21 am');
  });

  test('should return Time passed from now if format is set as fromNow', () => {
    const dt = new Date();
    dt.setMinutes(dt.getMinutes() - 2);
    const wrapper = mount(
      <FormattedDate format={'fromNow'}>{dt}</FormattedDate>,
    );
    expect(wrapper.find('time').text()).toBe('2 minutes ago');
  });
});
