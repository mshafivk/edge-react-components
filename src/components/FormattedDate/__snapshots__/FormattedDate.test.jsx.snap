// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Formatted Date Component should return Date and Time value if format is set as dateOnly 1`] = `
<FormattedDate
  format="fullDateTime"
>
  <n
    ago={false}
    calendar={false}
    decimal={false}
    element={null}
    filter={[Function]}
    format="MMMM Do YYYY, h:mm:ss a"
    fromNow={false}
    interval={60000}
    local={false}
    onChange={[Function]}
    titleFormat=""
    toNow={false}
    trim={false}
    unit={null}
    unix={false}
    utc={false}
    withTitle={false}
  >
    <time
      dateTime={"2021-02-25T01:11:21.000Z"}
    >
      February 25th 2021, 6:41:21 am
    </time>
  </n>
</FormattedDate>
`;

exports[`Formatted Date Component should return Date only value if format is set as dateOnly 1`] = `
<FormattedDate
  format="dateOnly"
>
  <n
    ago={false}
    calendar={false}
    decimal={false}
    element={null}
    filter={[Function]}
    format="DD/MM/YYYY"
    fromNow={false}
    interval={60000}
    local={false}
    onChange={[Function]}
    titleFormat=""
    toNow={false}
    trim={false}
    unit={null}
    unix={false}
    utc={false}
    withTitle={false}
  >
    <time
      dateTime={"2021-01-09T01:11:21.000Z"}
    >
      09/01/2021
    </time>
  </n>
</FormattedDate>
`;

exports[`Formatted Date Component should return Time passed from now if format is set as fromNow 1`] = `
<FormattedDate
  format="fromNow"
>
  <n
    ago={false}
    calendar={false}
    decimal={false}
    element={null}
    filter={[Function]}
    format={false}
    fromNow={true}
    interval={60000}
    local={false}
    onChange={[Function]}
    titleFormat=""
    toNow={false}
    trim={false}
    unit={null}
    unix={false}
    utc={false}
    withTitle={false}
  >
    <time
      dateTime={"2021-01-15T14:22:04.640Z"}
    >
      2 minutes ago
    </time>
  </n>
</FormattedDate>
`;
